integration/modeshape-jbossas-integration-tests/src/test/java/org/modeshape/test/integration/EsIndexIntegrationTest.java

```diff
@@ -0,0 +1,147 @@
+
+import java.io.File;
+import java.io.InputStream;
+import javax.annotation.Resource;
+import javax.jcr.Node;
+import javax.jcr.Session;
+import javax.jcr.query.Query;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.NodeBuilder;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ArchivePaths;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.junit.After;
+import org.junit.AfterClass;
+//import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+//import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.modeshape.common.util.FileUtil;
+import org.modeshape.jcr.JcrRepository;
+import org.modeshape.jcr.ValidateQuery;
+import org.modeshape.jcr.ValidateQuery.ValidationBuilder;
+import org.modeshape.jcr.api.JcrTools;
+
+/**
+ * Arquillian integration tests that uses the predefined repository that
+ * contains some index definitions, to test that query functionality is as
+ * expected.
+ */
+@RunWith(Arquillian.class)
+public class EsIndexIntegrationTest {
+
+    @Resource(mappedName = "/jcr/repo-with-es-index")
+    private JcrRepository repository;
+    private Session session;
+    private boolean print;
+    private JcrTools tools;
+    private static org.elasticsearch.node.Node esNode;
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        File[] testDeps = Maven.configureResolver()
+                .workOffline()
+                .loadPomFromFile("pom.xml")
+                .resolve("org.modeshape:modeshape-jcr:test-jar:tests:?").withTransitivity().asFile();
+
+        return ShrinkWrap.create(WebArchive.class, "query-test.war").addAsLibraries(testDeps)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, ArchivePaths.create("beans.xml"))
+                .addAsWebInfResource(new File("src/main/webapp/WEB-INF/jboss-deployment-structure.xml"), ArchivePaths.create("jboss-deployment-structure.xml"))
+                .addAsResource(new File("src/test/resources/text-extractor/text-file.txt"))
+                .setManifest(new File("src/main/webapp/META-INF/MANIFEST.MF"));
+    }
+
+    @BeforeClass
+    public static void startES() throws Exception {
+        FileUtil.delete("target/data");
+        Settings localSettings = Settings.settingsBuilder()
+                .put("http.enabled", true)
+                .put("number_of_shards", 1)
+                .put("number_of_replicas", 1)
+                .put("path.home", "target/data")
+                .build();
+
+        //configure Elasticsearch node
+        esNode = NodeBuilder.nodeBuilder().settings(localSettings).local(false).build().start();
+        Thread.currentThread().sleep(3000);
```

Comment:
I think 1000 (1sec) is enough here since this is the same value that the standalone tests use (from the es-index-provoider module)